Communications/Server.cs

```diff
@@ -1,61 +1,90 @@
 ﻿using Scarlet.Utilities;
+
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Sockets;
 using System.Threading;
-using System.Threading.Tasks;
 
 namespace Scarlet.Communications
 {
     public static class Server
     {
         private static Dictionary<string, ScarletClient> Clients;
-        private static Dictionary<string, Queue<Packet>> SendQueues;
-        private static Queue<Packet> ReceiveQueue;
+
+        // Buffer
+        private static Dictionary<string, PacketBuffer> SendQueues;
+        private static QueueBuffer ReceiveQueue;
+
+        // Port listener
         private static UdpClient UDPListener;
         private static TcpListener TCPListener;
+
+        // Threads
         private static Thread SendThread;
         private static Thread ReceiveThreadTCP, ReceiveThreadUDP, ProcessThread;
+
+        // Running status
         private static bool Initialized = false;
         private static volatile bool Stopping = false;
-        public static bool StorePackets = false;
-        public static List<Packet> PacketsReceived, PacketsSent;
         private static int ReceiveBufferSize, OperationPeriod;
+
+        // Watchdog
         public static event EventHandler<EventArgs> ClientConnectionChange;
         public static bool OutputWatchdogDebug = false;
 
+        // Packet dumping
+        public static bool StorePackets = false;
+        public static List<Packet> PacketsReceived, PacketsSent;
+
+        // Packet Priority
+        public static bool PriorityQueueEnabled { get; private set; }
+        public static PacketPriority DefaultPriority { get; private set; }
+
         /// <summary>
         /// Prepares the server for use, and starts listening for clients.
         /// </summary>
         /// <param name="PortTCP">The port to listen on for clients communicating via TCP.</param>
         /// <param name="PortUDP">The port to listen on for clients communicating via UDP.</param>
         /// <param name="ReceiveBufferSize">The size, in bytes, of the receive data buffers. Increase this if your packets are longer than the default.</param>
         /// <param name="OperationPeriod">The time, in ms, between network operations. If you are sending/receiving a lot of packets, and notice delays, lower this.</param>
-        public static void Start(int PortTCP, int PortUDP, int ReceiveBufferSize = 64, int OperationPeriod = 20)
+        public static void Start(int PortTCP, int PortUDP, bool UsePriorityQueue = true, int ReceiveBufferSize = 64, int OperationPeriod = 20)
```

Comment:
Please set this to false by default. Since most subsystems are not going to need priority-based sending, it makes more sense to turn it off by default, only enabling it when they do need it.

Reply:
Since I asked for UsePriorityQueue on client to be at the end of the method footprint, this one should probably be at the end too, as long as that's okay with you @CaiB

Reply:
Yup, that sounds like a good idea. Will reduce the number of things that will break, as some people like to use the parameters without naming them.